packages/contracts-sdk/src/implementations/farcaster/FarcasterIdRegistryContract.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  EVMContractAddress,
+  BlockchainCommonErrors,
+  FarcasterIdRegistryContractError,
+  FarcasterId,
+} from "@snickerdoodlelabs/objects";
+import { ethers } from "ethers";
+import { injectable } from "inversify";
+import { ResultAsync, errAsync, okAsync } from "neverthrow";
+
+import { BaseContract } from "@contracts-sdk/implementations/BaseContract.js";
+import { IEthersContractError } from "@contracts-sdk/implementations/BlockchainErrorMapper.js";
+import { IFarcasterIdRegistryContract } from "@contracts-sdk/interfaces/index.js";
+import { ContractsAbis } from "@contracts-sdk/interfaces/objects/index.js";
+
+@injectable()
+export class FarcasterIdRegistryContract
+  extends BaseContract<FarcasterIdRegistryContractError>
+  implements IFarcasterIdRegistryContract
+{
+  constructor(
+    protected providerOrSigner: ethers.Provider | ethers.Signer,
+    protected contractAddress: EVMContractAddress,
+  ) {
+    super(
+      providerOrSigner,
+      contractAddress,
+      ContractsAbis.FarcasterIdRegistryAbi.abi,
+    );
+  }
+
+  public idOf(
+    ownerAddress: EVMContractAddress,
+  ): ResultAsync<
+    FarcasterId,
```

Comment:
There's a type "FarcasterUserId" already defined in the IP repo; I recommend moving that over to here. We can refactor the IP to use the one from the protocol later.